pymagicc/config.py

```diff
@@ -0,0 +1,102 @@
+"""
+Module for collating configuration variables from various sources
+
+The order of preference is:
+Environment variable > User configuration file > Defaults
+"""
+
+import platform
+from os import environ
+from os.path import abspath, dirname, exists, expanduser, join
+
+import yaml
+
+__all__ = ['config']
+
+_is_windows = platform.system() == "Windows"
+# Default configuration parameters for pymagicc
+default_config = {
+    'EXECUTABLE': join(dirname(abspath(__file__)),
+                       "MAGICC6/MAGICC6_4Download/magicc6.exe"
+                       ),
+    'WINDOWS': _is_windows
+}
+
+_xdg_home = environ.get('XDG_CONFIG_HOME', expanduser('~/.config/'))
+
+
+def user_config_file():
+    """
+    Get the file name for the user-level config file.
+
+    This configuration file is stored in the User profile and contains any user
+    specific overrides to pymagicc configuration. Under Windows this file will
+    be named `%USERPROFILE%/.pymagicc.yml` for other OS the file should be
+    located at `~/.config/pymagicc.yml`. This file is not created by default
+    """
+    if _is_windows:
+        return expanduser('~\\.pymagicc.yml')
+    return join(_xdg_home, 'pymagicc.yml')
+
+
+def parse_config_file(fname):
+    """
+    Read in a configuration file from a given filename
+    :param fname:
+    :return: The configuration in the file as a Dict, else None if the file
+        could not be found
+    """
+    if exists(fname):
+        data = yaml.load(open(fname))
+        # Convert all keys to uppercase
```

Comment:
is this just a convenience to avoid dealing with case-sensitivity?

Reply:
It was for testing. It enables this whole function to be replaced with the expected loaded data. Otherwise, you have to mock open to return a stringified version of the configuration and it gets complicated fast!  See line 47

Reply:
ah ok I see